function/authorization.js

```diff
@@ -30,6 +53,54 @@ class GoogleOAuthAuthorizer extends Authorizer {
         return persistentRequest.post(reqOptions)
     }
 
+    validateAudienceOrEmail(audience, email) {
+        // following two loops use for..in instead of Array.forEach to allow breaking out early
+        // does the audience value start with one of our whitelisted prefixes?
+        for (const i in this.audiencePrefixes) {
+            const prefix = this.audiencePrefixes[i];
+            if (audience.startsWith(prefix.toString())) {
+                return true;
+            }
+        }
+        // does the email value end with one of our whitelisted suffixes?
+        for (const i in this.emailSuffixes) {
+            const suffix = this.emailSuffixes[i];
+            if (email.endsWith(suffix.toString())) {
+                return true;
+            }
+        }
+
+        return false;
```

Comment:
I can tell this implements the javascript equivalent of the python from bond:
`any(audience.startswith(prefix) for prefix in self.config.accepted_audience_prefixes) or any(token_info.get('email').endswith(suffix) for suffix in self.config.accepted_email_suffixes)`

but the early returns hide this. I don't think the early returns buy us anything. Is there a javascript equivalent that can avoid them?

Reply:
the early returns are there to short-circuit out of the loops so the code doesn't do extra work - same way the `any` would. If you think it would be clearer, I could rewrite the loops with `break` instead of `return` - is that what you're after?